Fix decomposition for std

lezcano opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #87182
* #87183
* #87205
* #87204
* #87203
* #87424
* __->__ #87181
* #87282

The previous implementation was lacking a few features and incurred on a
pretty large error

cc @ezyang @mruberry @ngimel @Lezcano @fdrocha

lezcano committed:
Fix decomposition for std

The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

Chillee commented:
Hmm... this decomposition *is* notably different than the previous one. While the previous decomposition decomposed var in terms of mean and sum, this one decomposes it into `prim.var`.

ngimel commented:
Yeah but inductor has its own var lowering. I'm not sure if functorch is going to be fine with prims.var

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

[ghstack-poisoned]

lezcano committed:
Update on "Fix decomposition for std"


The previous implementation was lacking a few features and incurred on a
pretty large error

cc ezyang mruberry ngimel Lezcano fdrocha

[ghstack-poisoned]

lezcano commented:
@pytorchbot merge
